### ISSUE TITLE
[Bug] Freecam Bug

### ISSUE THREAD

**Issue by RickyTheRacc** [MeteorDevelopment/meteor-client#1877]
### Describe the bug

Freecam will still move around if the movement keys are pressed in chat and in guis

### Steps to reproduce

```bash
Turn on freecam and open a chest or furnace, and shift click an item into or out of it. Notice that your camera will move downwards as you shift click, despite being in a gui.
```

### Meteor Version

1384

### Minecraft Version

1.18.1

### Operating System

Windows

### Before submitting a bug report

- [X] This bug wasn't already reported.
(I have checked every bug report on github)

- [X] This is a valid bug.
(I am able to reproduce this on the latest dev build)

**Comment by TheRedHippo** [comment]
you got gui move on?

**Comment by arlomcwalter** [comment]
^

**Comment by arlomcwalter** [comment]
@RickyTheRacc

**Comment by RickyTheRacc** [comment]
Sheeeeeiiiiittttt

**Comment by SIMULATAN** [comment]
This might still apply tho, people might want to use InvWalk but don't want to walk when in freecam.
Also I noticed that walking works in chat too which is a bit annoying, maybe we can add a option for that

**Comment by RickyTheRacc** [comment]
Actually cope, it still happens without inv move on

**Comment by wsquarepa** [comment]
Happening for me with the chat window as well
GUI move is NOT on

**Comment by MineGame159** [comment]
Fixed in d090e84a82f1a62c3bc8c53f5a17072309d3dc40.
